core/iwasm/libraries/lib-wasi-threads/lib_wasi_threads_wrapper.c

```diff
@@ -130,9 +182,12 @@ thread_spawn_wrapper(wasm_exec_env_t exec_env, uint32 start_arg)
     return thread_id;
 
 thread_spawn_fail:
+    os_mutex_unlock(&exec_env->wait_lock);
```

Comment:
Not sure why moving `os_mutex_unlock` to here, putting it in the original place (L177) seems to be better?

Reply:
To separate the error handling logic. It's even more useful if in the future we put additional operations between L179 and L180 and we want to release the lock in case of errors.